aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/IntegrationTests/TestFixture.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace TestingControllersSample.Tests.IntegrationTests
+{
+    public class TestFixture<TStartup> : IDisposable
+    {
+        private const string SolutionName = "TestingControllersSample.sln";
+        private readonly TestServer _server;
+
+        public TestFixture()
+            : this(Path.Combine("src"))
+        {
+        }
+
+        protected TestFixture(string solutionRelativePath)
+        {
+            var startupAssembly = typeof(TStartup).GetTypeInfo().Assembly;
+            var contentRoot = GetProjectPath(solutionRelativePath, startupAssembly);
+
+            var builder = new WebHostBuilder()
+                .UseContentRoot(contentRoot)
```

Comment:
I couldn't quite tell - but do we do views tests in this sample? If there are only API tests, we could drop this code (and finding the sln snippet). Alternatively add a test for views (assuming it's not already there)